pandas/tests/frame/test_duplicates.py

```diff
@@ -263,6 +263,21 @@ def test_drop_duplicates_tuple():
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize('df', [
+    DataFrame(),
+    DataFrame(columns=[]),
+    DataFrame(columns=['A', 'B', 'C']),
+    DataFrame(index=[]),
+    DataFrame(index=['A', 'B', 'C'])
+])
+def test_drop_duplicates_empty(df):
+    # GH 20516
+    result = df.drop_duplicates()
+    if df.columns.empty is False:
+        df = DataFrame(columns=[])
+    tm.assert_frame_equal(result, df)
```

Comment:
This tests feels a bit tricky to me. So, when there are columns, we set them to an empty list. So, that would be the same as simply using as expected value `DataFrame(columns=[])`, or am I missing something?

But not sure if I'm missing something, but I'd expect `DataFrame(columns=['A', 'B', 'C']).drop_duplicates()` to keep the columns. Any reason to drop them?

Reply:
Please check the closed issue of #22409. I've been told that it is an expected behavior.

Reply:
I don't think the problem may be your approach implementing this story. I agree on what @WillAyd said in #22409, but not in the side effect in this PR.

I guess `df[pandas.Series()]` is the same as `df[[]]`, so you're telling the DataFrame to select an empty list of columns, so none of them. So, if we have a DataFrame with 4 columns and 10 rows, I expect this to return the 10 rows, but 0 columns.

But the case here with `drop_duplicates`, if I understand correctly, is that you have a DataFrame with 4 columns and 0 rows, you remove the duplicates, that there are none, as the DataFrame has 0 rows, so I should still have 4 columns and 0 rows.

And in your test you're asserting that the return will be 0 columns and 0 rows.

Am I right? Does it makes sense?

Reply:
Yeah. However, if we want the column to carry over, then we'd need to change the drop_duplicates itself to not use the logic of df[~duplicated], as it currently does, as this is what is causing this problem at least at this moment with the empty return of duplicated... Maybe returning empty series by itself wasn't a good idea. Completely stuck then.